packages/navigator/src/browser/navigator-filter.ts

```diff
@@ -23,9 +23,13 @@ export class FileNavigatorFilter {
 
     protected filterPredicate: FileNavigatorFilter.Predicate;
 
+    protected showHiddenFilesFlag: boolean;
+    protected exclusions: FileNavigatorFilter.Exclusions | undefined;
```

Comment:
We could get rid of this field, cannot we? We need it once at construction time and once when the user toggles the state, no?

Reply:
Agree, but I think, I can't get the main idea of your comment.

Reply:
If you would get rid of this field, you would need to modify the toggle method to something like this:

```typescript
toggleHiddenFiles(): void {
  this.showHiddenFiles = !this.showHiddenFiles;
  this.filterPredicate = this.createFilterPredicate(this.preferences['navigator.exclude'] || {});
  this.fireFilterChanged();
}
```

And obviously, you would not need to reassign the new value to the `exclusions` field in the constructor and in the `onPreferenceChanged` method. Does this answer your question?

Reply:
Get it! I've refactored code and removed redundant variable. Thanks for pointing!